### PR TITLE
chore: generate dependabot configs for examples

### PR DESCRIPTION
**Pull request by mdelapenya**
## What does this PR do?
It models dependabot config file, using it to generate dependabot configs for each generated example module.

This generation has this requirements:
- the main module comes first in dependabot.yml
- the existing e2e module comes second
- all the examples comes after, in alphabetical order. This is provided by the `sort.Sort` function, therefore the Updates struct implements the Len, Less and Swap methods of the Sort interface. We can sort updates by its directory.

## Why is it important?
Update example modules with dependencies in an automated manner

## Related issues

- Completes #618

### CODE REVIEW

**Review by eddumelendez — COMMENTED** [review]
besides adding the new section for the new example is removing the quotes to the existing ones

Screenshot 2022-12-05 at 11 59 00

**Comment by mdelapenya** [comment]
Interesting 🤔 let me check how the YAML library is marshaling the strings, as it should include the quotes

**Comment by eddumelendez** [comment]
I think we should remove them first in `main` branch. So, adding a new example is just update the file with the right changes. WDYT?

File: .github/dependabot.yml [hunk 1]
```diff
@@ -12,3 +12,57 @@ updates:
     interval: daily
   open-pull-requests-limit: 3
   rebase-strategy: disabled
+- package-ecosystem: gomod
+  directory: "/examples/cockroachdb"
```
Comment (eddumelendez) [inline .github/dependabot.yml:16]: remove quotes :)

```suggestion
  directory: /examples/cockroachdb
```

Comment (mdelapenya) [reply .github/dependabot.yml]: Removed in [a401b2d](https://github.com/testcontainers/testcontainers-go/pull/654/commits/a401b2d8f471173ec9b6a932ba56f5aeab39f332)

**Review by eddumelendez — COMMENTED** [review]
when adding a new example the format change. Once this is fixed we are ready to go!

Screenshot 2022-12-05 at 12 30 40

**Comment by mdelapenya** [comment]
@eddumelendez I think go-yaml is still producing valid indents for a YAML file even though the original indents are not honored. Please see https://github.com/go-yaml/yaml/issues/779

**Review by eddumelendez — APPROVED** [review]

**Pull request merged**